### PR TITLE
User upgrades and base repos upgrades and fixes

### PR DESCRIPTION
**Pull request by mszynka**

### CODE REVIEW

File: TaskVlopper.Repository/MeetingParticipantsRepository.cs [hunk 1]
```diff
@@ -12,17 +12,27 @@ public class MeetingParticipantsRepository : BaseRepository<MeetingParticipants>
...
+
+        public IQueryable<MeetingParticipants> GetMeetingParticipantsByUserIdQueryable(string userId)
+        {
             return this.GetAll().Where(x => x.UserID == userId);
         }
 
         public MeetingParticipants GetMeetingParticipantsByUserIdAndMeetingId(string userId, int meetingId)
         {
-            return GetMeetingParticipantsByUserIdAndMeetingIdQueryable(userId, meetingId).Single();
+            return this.GetAll().Where(x => x.UserID == userId && x.MeetingID == meetingId).Single();
```
Comment (mszynka) [inline TaskVlopper.Repository/MeetingParticipantsRepository.cs]: Can I say I've done it more... properly?

File: TaskVlopper.Repository/UserTaskAssignmentRepository.cs [hunk 2]
```diff
@@ -12,37 +12,58 @@ public class UserTaskAssignmentRepository : BaseRepository<UserTaskAssignment>,
...
         public IEnumerable<UserTaskAssignment> GetTaskAssignmentByUserIdAndTaskId(string userId, int taskId)
         {
-            return this.GetAll().Where(x => x.TaskID == taskId && x.UserID == userId);
+            return GetTaskAssignmentByUserIdAndTaskId(userId, taskId).AsEnumerable();
         }
 
         public UserTaskAssignment GetTaskAssignmentByUserIdAndProjectIdAndTaskId(string userId, int projectId, int taskId)
         {
-            return this.GetTaskAssignmentByUserIdAndProjectIdAndTaskIdQueryable(userId, projectId, taskId).Single();
+            return this.GetAll().Where(x => x.ProjectID == projectId && x.TaskID == taskId && x.UserID == userId).Single();
```
Comment (mszynka) [inline TaskVlopper.Repository/UserTaskAssignmentRepository.cs]: I'd also use queryable here as execution and then Single() on it. It's Queryable anyway

File: TaskVlopper.Repository/UserTaskAssignmentRepository.cs [hunk 3]
```diff
@@ -12,37 +12,58 @@ public class UserTaskAssignmentRepository : BaseRepository<UserTaskAssignment>,
...
+            return GetTaskAssignmentByUserIdAndProjectIdQueryable(userId, projectId).AsEnumerable();
         }
 
         public IEnumerable<string> GetAllUsersIDsForGivenTaskProject(int projectId, int taskId)
         {
-            return this.GetAll().Where(x => x.ProjectID == projectId && x.TaskID == taskId).Select(x => x.UserID);
+            return GetAllUsersIDsForGivenTaskProjectQueryable(projectId, taskId).AsEnumerable();
+        }
+
+        //Queryables
```
Comment (mszynka) [inline TaskVlopper.Repository/UserTaskAssignmentRepository.cs]: Region Queryables perhaps would be nicer

File: TaskVlopper/Controllers/MeetingController.cs [hunk 4]
```diff
@@ -184,9 +185,11 @@ public ActionResult Users(int id)
                 if (User.Identity.IsAuthenticated)
                 {
                     IMeetingLogic logic = container.Resolve<IMeetingLogic>();
-                    var viewmodel = logic.GetMeetingUsers(id).ToList();
+                    var queryMeetingUsers = logic.GetMeetingUsers(id);
 
-                    return Json(viewmodel, JsonRequestBehavior.AllowGet);
+                    var viewModel = new UsersViewModel();
+                    viewModel.Users.AddRange(queryMeetingUsers.Select(x => new UserViewModel(x)));
```
Comment (mszynka) [inline TaskVlopper/Controllers/MeetingController.cs:265]: I've added overloaded constructor that can do AddRange. In ProjectViewModel but that's a baseline mostly

**Comment by bojakowsky** [comment]
Bugs on FrontEnd detected:
- Project updates are not working,
- Worklog, Tasks, Meetings are not working
- Generally, front end gone a bit crazy
- Calendars on tasks are working wrongly
- Statistics are not wroking on project nor any of others

If you smarten up the front end bugs, I'll check it, and then we can merge.

**Comment by mszynka** [comment]
Front end:
- Projects fixed, commit comming
- What you mean by not working? Be more specific
- Generally front end is crazy cause I don't get everything in my head
- Calendars were never working properly :disappointed: 
- Statistics not working? Maybe check what they are and what they are doing before blame :wink:

**Comment by bojakowsky** [comment]
project run sucessfully after clean branch checkout

**Comment by mszynka** [comment]
@bojakowsky Yeah. It should. I had to remove some commits but ultimately it has to be working :smile:

**Pull request merged**